### PR TITLE
EXP Behavior Changes

### PR DESCRIPTION
**Pull request by Jedzkie**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** master

[//]: # (Master? Slave?)

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1703)

**Comment by Smokexyz** [comment]
It would be easier to understand the changes you intend to make in the PR if you describe them in the PR form. E.g. Why would job exp received by quests not be shown in purple?

**Comment by Smokexyz** [comment]
It would be easier for us to understand the changes you intend to make if you fill out the description part of the PR form. E.g. I don't understand why job exp received from quests wouldn't be purple. I remember this this happened in iRO.

File: src/map/pc.c [hunk 1]
```diff
@@ -6750,16 +6750,17 @@ int pc_checkjoblevelup(struct map_session_data *sd)
 /**
  * Alters EXP based on self bonuses that do not get shared with the party
  **/
-void pc_calcexp(struct map_session_data *sd, unsigned int *base_exp, unsigned int *job_exp, struct block_list *src) {
+void pc_calcexp(struct map_session_data *sd, unsigned int *base_exp, unsigned int *job_exp, struct block_list *src)
+{
 	int buff_ratio = 0, buff_job_ratio = 0, race_ratio = 0, pk_ratio = 0;
```
Comment (Smokexyz) [inline src/map/pc.c]: Wouldn't changing the 1 to 0 do the same thing you intend to do here?

File: src/map/pc.c [hunk 2]
```diff
@@ -6796,27 +6797,31 @@ void pc_calcexp(struct map_session_data *sd, unsigned int *base_exp, unsigned in
...
+		jexp = apply_percentrate64(jexp, sd->status.mod_exp, 100);
 	}
 
 	bexp += apply_percentrate64(bexp, buff_ratio, 100);
 	jexp += apply_percentrate64(jexp, buff_ratio + buff_job_ratio, 100);
 
-	*job_exp = (unsigned int)cap_value(jexp, 1, UINT_MAX);
-	*base_exp = (unsigned int)cap_value(bexp, 1, UINT_MAX);
+	if (*base_exp) {
+		*base_exp = (unsigned int)cap_value(bexp, 1, UINT_MAX);
```
Comment (Smokexyz) [inline src/map/pc.c]: Wouldn't changing the 1 to 0 do the same thing you intend to do here?

Comment (Jedzkie) [inline src/map/pc.c]: Done.

**Comment by MishimaHaruna** [comment]
"If the monster has no EXP, it will not give 1 EXP when killed."

Exp can be zero for other reasons as well (example: a character deals 1 damage to a monster, or uses Lex Aetherna, then another character kills it). Should these cases also give 0 exp? I'm used to see 1 in those cases, was that an incorrect behavior?

**Comment by Jedzkie** [comment]
Yes, it's incorrect behavior, There's a FABRE that only gives exp, i checked its info in our db and it has 0 jexp, when i killed it in-game, it gives me 1 jexp. which is very wrong.

**Pull request closed without merging**